packages/modal/src/useModal.ts

```diff
@@ -9,18 +9,37 @@ import { useState } from 'react';
 import { composeEventHandlers, generateId, KEY_CODES } from '@zendeskgarden/container-utilities';
 import { useFocusJail } from '@zendeskgarden/container-focusjail';
 
-export function useModal({ onClose, modalRef, id: _id, focusOnMount, environment } = {}) {
+export interface IUseModalProps {
+  onClose?: (event: Event) => any;
+  modalRef: React.RefObject<HTMLElement>;
+  id?: string;
+  focusOnMount?: boolean;
+  environment?: Document;
+}
+
+export interface IUseModalReturnValue {
+  getBackdropProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getModalProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getTitleProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getContentProps: <T>(options?: T) => T & React.HTMLProps<any>;
+  getCloseProps: <T>(options?: T) => T & React.ButtonHTMLAttributes<any>;
+  closeModal?: <T>(event: T & Event) => void;
+}
+
+export function useModal(
+  { onClose, modalRef, id: _id, focusOnMount, environment }: IUseModalProps = {} as any
+): IUseModalReturnValue {
   const [idPrefix] = useState(_id || generateId('garden-modal-container'));
   const titleId = `${idPrefix}--title`;
   const contentId = `${idPrefix}--content`;
 
-  const closeModal = event => {
+  const closeModal = (event: Event) => {
```

Comment:
Let's limit this to `KeyboardEvent | MouseEvent`

Reply:
Updated, thanks!